### PR TITLE
358 fix/delete posts by post status

### PR DESCRIPTION
**Pull request by rajanit2000**
Blocked by #332

### CODE REVIEW

**Comment by sudar** [comment]
@rajanit2000 

This branch has lot of additional changes as well. Didn't you created a seperate branch for this issue?

There are lot of codacy comments. Can you fix them as well and submit it for PR again?

**Comment by rajanit2000** [comment]
@sudar i have create new branch and create new pull request for this issue #407

**Comment by rajanit2000** [comment]
Its just like #332 
Codacy still is not passed, But want to review
Codecy issue something we want to ignore it
https://app.codacy.com/app/rajanit2000/bulk-delete/pullRequest?prid=1892658

**Comment by rajanit2000** [comment]
Development is done, Ready to review

File: tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php [hunk 1]
```diff
@@ -334,6 +477,39 @@ public function test_that_draft_posts_that_are_posted_within_the_last_x_days_can
...
+		$delete_options = array(
+			'post_status'  => array( 'draft' ),
+			'limit_to'     => -1,
+			'restrict'     => true,
+			'force_delete' => true,
+			'date_op'      => 'after',
+			'days'         => '5',
+		);
+
+		$posts_deleted = $this->module->delete( $delete_options );
```
Comment (sudar) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php]: Assert that some additional posts that don't match the condition are not deleted.

Do this for all the other test cases as well.

**Comment by sudar** [comment]
@rajanit2000 

Make sure you are adding some asserts to check that posts that don't satisfy the condition are not getting deleted.

Also pull the latest code from github.

File: tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php (hunk 1 above)
Comment (rajanit2000) [reply tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php]: Fixed

**Comment by rajanit2000** [comment]
@sudar its ready to review

File: tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php [hunk 2]
```diff
@@ -30,21 +30,32 @@ public function setUp() {
...
+		$this->factory->post->create_many(
+			10, array(
+				'post_type'   => 'post',
+				'post_status' => 'pending',
+			)
+		);
+
 		$delete_options = array(
 			'post_status'  => array( 'publish', 'draft' ),
 			'limit_to'     => -1,
```
Comment (rajanit2000) [inline tests/wp-unit/include/Core/Posts/Modules/DeletePostsByStatusModuleTest.php]: Check  limit_to value

**Pull request merged**